File: ros2-py-outputs/agr/autonomous_ground_robot/obstacle_detector_node.py
"""ObstacleDetector — node implementation. EDIT ME.

Generated ONCE by the ros2-sysmlv2 bridge pipeline and never overwritten
(generation-gap pattern): this file is where demo/application logic lives.
The architectural wiring (endpoints, parameters, lifecycle transitions) is in
obstacle_detector_node_base.py, which is regenerated on every pipeline run.
"""
import rclpy

from .obstacle_detector_node_base import ObstacleDetectorBase


class ObstacleDetector(ObstacleDetectorBase):
    """obstacle_detector application logic."""

    def handle_rawSub(self, msg) -> None:
        """/scan (LaserScan).

        TODO: implement message processing logic.
        """

    def handle_cameraSub(self, msg) -> None:
        """/camera/image (Image).

        TODO: implement message processing logic.
        """


def main(args=None):
    rclpy.init(args=args)
    node = ObstacleDetector()
    rclpy.spin(node)
    node.destroy_node()
    rclpy.shutdown()


if __name__ == '__main__':
    main()

File: ros2-cpp-outputs/agr/src/obstacle_detector_node.cpp
// ObstacleDetector — node implementation. EDIT ME.
//
// Generated ONCE by the ros2-sysmlv2 bridge pipeline and never overwritten
// (generation-gap pattern): this file is where demo/application logic lives.
// The architectural wiring is in obstacle_detector_node_base.hpp/.cpp, regenerated
// on every pipeline run.
#include "autonomous_ground_robot/obstacle_detector_node_base.hpp"

class ObstacleDetector : public ObstacleDetectorBase
{
public:
  using ObstacleDetectorBase::ObstacleDetectorBase;

protected:
  void handle_rawSub(const sensor_msgs::msg::LaserScan & msg) override
  {
    // /scan — TODO: implement message processing logic.
    (void)msg;
  }
  void handle_cameraSub(const sensor_msgs::msg::Image & msg) override
  {
    // /camera/image — TODO: implement message processing logic.
    (void)msg;
  }
};

int main(int argc, char ** argv)
{
  rclcpp::init(argc, argv);
  auto node = std::make_shared<ObstacleDetector>();
  rclcpp::spin(node->get_node_base_interface());
  rclcpp::shutdown();
  return 0;
}
